product_stock_state/models/product_product.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2017-Today GRAP (http://www.grap.coop).
+# Copyright 2018 ACSONE SA/NV
+# Copyright 2018 Akretion (http://www.akretion.com).
+# @author Sylvain LE GAL <https://twitter.com/legalsylvain>
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# @author Laurent Mignon <laurent.mignon@acsone.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    _STOCK_STATE_SELECTION = [
+        ("in_stock", "In Stock"),
+        ("in_limited_stock", "In Limited Stock"),
+        ("resupplying", "Resupplying"),
+        ("out_of_stock", "Out Of Stock"),
+    ]
+
+    stock_state = fields.Selection(
+        selection=_STOCK_STATE_SELECTION, compute="_compute_stock_state"
+    )
+
+    def _get_qty_available(self):
```

Comment:
To avoid to have 2 unrelated methods which conflicts on their name (low probability but ...)
```suggestion
    def _get_qty_available_for_stock_state(self):
```